File: API-Biblioteca-DevEficiente-CDD/src/main/java/io/github/monthalcantara/bibliotecaCDD/usuario/NovoUsuarioRequest.java
package io.github.monthalcantara.bibliotecaCDD.usuario;


import javax.validation.constraints.NotNull;

public class NovoUsuarioRequest {

    @NotNull
    private String tipoUsuario;

    @Deprecated
    private NovoUsuarioRequest() {
    }

    public NovoUsuarioRequest(String tipoUsuario) {
        this.tipoUsuario = tipoUsuario;
    }

    public Usuario toModel(){
        TipoUsuario usuario = TipoUsuario.toEnum(this.tipoUsuario);
        return new Usuario(usuario);
    }

    public String getTipoUsuario() {
        return tipoUsuario;
    }
}

File: Kotlin/API-Casa_Do_Codigo_Kotlin/src/main/kotlin/io/github/monthalcantara/casadocodigo/dto/request/usuario/NovoUsuarioRequest.kt
package io.github.monthalcantara.casadocodigo.dto.request.usuario

import io.github.monthalcantara.casadocodigo.model.Usuario
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank

data class NovoUsuarioRequest(
    @field:NotBlank @field:Email val username: String,
    @field:NotBlank val password: String
) {
    fun toModel(): Usuario {
        val bcript = BCryptPasswordEncoder().encode(this.password)
        return Usuario(username = this.username, password = bcript)
    }
}
